kolibri/core/device/kolibri_plugin.py

```diff
@@ -0,0 +1,22 @@
+from kolibri.core.auth.hooks import FacilityDataSyncHook
+from kolibri.plugins.hooks import register_hook
+
+
+@register_hook
+class SingleUserExamSyncHook(FacilityDataSyncHook):
+    def pre_transfer(
+        self,
+        dataset_id,
+        local_is_single_user,
+        remote_is_single_user,
+        single_user_id,
+        context,
+    ):
+        # if we're about to send data to a single-user device, prep the syncable exam assignments
+        if context.sync_session and single_user_id is not None:
```

Comment:
I don't think this condition matches the comment; the condition I think just says something like "if either I or the other side is a single-user device, and one of us is about to send data to the other".

To match the comment, I think it'd be:
`if context.is_producer and remote_is_single_user:`

Reply:
The comment is also left over pasta.

Reply:
🍝